tests/general/LicenseDetectionBearTest.py

```diff
@@ -0,0 +1,66 @@
+from queue import Queue
+import os
+
+from bears.general.LicenseDetectionBear import LicenseDetectionBear
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+from coalib.testing.BearTestHelper import generate_skip_decorator
+from coalib.results.Result import Result
+from coalib.settings.Section import Section
+
+
+def get_testfile_path(name):
+    return os.path.join(os.path.dirname(__file__),
+                        'licensedetect_test_files',
+                        name)
+
+
+def load_testfile(name):
+    with open(get_testfile_path(name)) as fl:
+        contents = fl.read()
+
+    return contents
+
+
+@generate_skip_decorator(LicenseDetectionBear)
+class LicenseDetectionBearTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.uut = LicenseDetectionBear(Section('name'), Queue())
+
+    def test_license(self):
+        test_file_path = get_testfile_path('apache_license_2.py')
+        file_contents = [load_testfile(test_file_path)]
+        self.check_results(
+            self.uut,
+            file_contents,
+            [Result.from_values('LicenseDetectionBear',
+                                test_file_path + ': Apache (v2.0)',
```

Comment:
We will have to provide a number of options to the user(total available licenses that our bear can detect) and the user may not have that knowledge. I suggest that our bear should check whether the given file has a license or not. It will only complain if the file doesn't have a license. Something similar to: 
```
    def process_output(self, output, filename, file):
        stdout, stderr = output
        result = filename + ': *No copyright* UNKNOWN'
        if result.strip() == stdout.strip():
            yield from self.process_output_regex(stdout, filename, file,
                                                 self._output_regex,
                                                 result_message='No license'
                                                                ' found.')
```
where we process that output and check if the file has a license or not.
Here is the output : 
```
s.py
|    | [NORMAL] LicenseDetectionBear:
|    | No license found.
|    | *0: Do nothing
|    |  1: Open file(s)
|    |  2: Add ignore comment
|    | Enter number (Ctrl-D to exit): 0
```
We can thus rename it to LicenseCheckBear since it serves the purpose. :smile: